tests/unit/test_leo.py

```diff
@@ -249,11 +251,13 @@ def landsat_base_url() -> str:
 
 
 @pytest.fixture()
-def landsat_cachestore() -> cache.DiskCacheStore:
+def landsat_cachestore() -> Generator[cache.DiskCacheStore, None, None]:
     """Clearable cache for Landsat data."""
     cache_root = cache._get_user_cache_dir()
     cache_dir = f"{cache_root}/landsat-unit-test"
-    return cache.DiskCacheStore(cache_dir=cache_dir, allow_clear=True)
+    cachestore = cache.DiskCacheStore(cache_dir=cache_dir, allow_clear=True)
+    yield cachestore
+    cachestore.clear()
```

Comment:
This is a nice change!

There is probably some room to reduce the time it takes to run the LEO test suite by letting data persist in the cache across tests. For example, test_landsat_get_reflective_bands currently downloads each band three times (once for each "processing" value). I structured the tests this way so they'd be as independent as possible, but if they're painfully slow for you we could consider changes that reduce independence but improve performance.

Reply:
Yeah ... it's really just the download that is slow. We could take a similar approach that we use for some the notebooks, and not clear the cache between test runs.

I'll leave as is for now.